hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityConverter.java

```diff
@@ -78,13 +90,25 @@
                     entity,
                     getBaselineEntity(baselineEntityMap, entity.getId()),
                     edgeLookup.getIncoming().row(entity),
-                    edgeLookup.getOutgoing().row(entity)))
+                    edgeLookup.getOutgoing().row(entity),
+                    buildLabelResultSet(entity, request, labelsMap)))
         .toList()
         .map(
             entities ->
                 new ConvertedEntityResultSet(entities, response.getTotal(), entities.size()));
   }
 
+  private Map<String, String> getLabelsMap(Optional<GetLabelsResponse> labelsResponse) {
+    return labelsResponse
+        .map(GetLabelsResponse::getLabelsList)
+        .orElse(Collections.emptyList())
+        .stream()
+        .collect(
+            Collectors.toUnmodifiableMap(
+                org.hypertrace.label.config.service.v1.Label::getId,
+                org.hypertrace.label.config.service.v1.Label::getKey));
```

Comment:
Can we have the entire Label as the value in the map, instead of just the label key? The reason being the Label can have more fields in future, and we would then anyways have to change value in the map to Label

Reply:
makes sense

Reply:
It will still require some code change in any case.

Reply:
definitely, trying to keep the code changes confined to label to label result conversion, apart from the schema changes